deepchem/models/keras_model.py

```diff
@@ -312,16 +351,22 @@ def fit(self,
       If specified, all logged losses are appended into this list. Note that
       you can call `fit()` repeatedly with the same list and losses will
       continue to be appended.
+    metrics: Optional[List[Metric]], optional (default None)
+      metrics to compute on the dataset used during training. If None,
+      no metrics and scores will be computed and only training loss will be logged.
 
     Returns
     -------
     The average loss over the most recent checkpoint interval
    """
+    self.dataset = dataset
+    self.metrics = metrics
```

Comment:
This isn't going to work.  We can't assume the user calls `fit()` and passes in a dataset and metrics.  They might call `fit_generator()` directly.  In that case, there is no dataset and epochs are undefined.

Reply:
Is there a preferred way for us to calculate the epoch number? We would need this value for logging based on epoch interval. Currently, we are calculating it based on the size of the training dataset. But as you mentioned, the training dataset may not exist if they are calling `fit_generator()` directly and not calling `fit()`

It's not clear to me whether there is a good way to calculate the epoch number directly inside `fit_generator()`. We could just check to see if the user has started training via `fit()` and in that case they would have passed in a Dataset object.

If there isn't a clear way, we can just remove the epoch logging option entirely and go back to only supporting step intervals.